server/plugin.go

```diff
@@ -204,6 +208,79 @@ func (p *Plugin) OnActivate() error {
 	}
 
 	go p.initStats()
+
+	go func() {
+		time.Sleep(time.Second * 10)
+
+		instances, err := p.instanceStore.LoadKnownJIRAInstances()
+		if err != nil {
+			p.API.LogError("unable to register autolinks", "err", err)
+			return
+		}
+
+		for url := range instances {
+			instance, err := p.instanceStore.LoadJIRAInstance(url)
+			if err != nil {
+				continue
+			}
+
+			jci, ok := instance.(*jiraCloudInstance)
+			if !ok {
+				p.API.LogWarn("only cloud instances supported for autolink", "err", err)
+				continue
+			}
+
+			if err := p.AddAutolinksForCloudInstance(jci); err != nil {
+				p.API.LogWarn("could not install autolinks for cloud instance", "instance", jci.BaseURL, "err", err)
+				continue
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (p *Plugin) AddAutolinksForCloudInstance(jci *jiraCloudInstance) error {
+	client, err := jci.getJIRAClientForServer()
+	if err != nil {
+		return fmt.Errorf("unable to get jira client for server: %w", err)
+	}
+
+	keys, err := JiraClient{Jira: client}.GetAllProjectKeys()
+	if err != nil {
+		return fmt.Errorf("unable to get project keys: %w", err)
+	}
+
+	for _, key := range keys {
+		err = p.AddAutolinks(key, jci.BaseURL)
+	}
+	if err != nil {
+		return fmt.Errorf("some keys were not installed: %w", err)
+	}
+
+	return nil
+}
+
+func (p *Plugin) AddAutolinks(key, baseURL string) error {
+	baseURL = strings.TrimRight(baseURL, "/")
+	installList := []autolink.Autolink{
+		{
+			Name:     key + " key to link for " + baseURL,
+			Pattern:  `(` + key + `)(-)(?P<jira_id>\d+)`,
+			Template: `[` + key + `-${jira_id}](` + baseURL + `/browse/` + key + `-${jira_id})`,
+		},
+		{
+			Name:     key + " link to key for " + baseURL,
+			Pattern:  `(` + strings.ReplaceAll(baseURL, ".", `\.`) + `/browse/)(` + key + `)(-)(?P<jira_id>\d+)`,
+			Template: `[` + key + `-${jira_id}](` + baseURL + `/browse/` + key + `-${jira_id})`,
+		},
+	}
+
+	client := autolinkclient.NewClientPlugin(p.API)
+	if err := client.Add(installList...); err != nil {
```

Comment:
@crspeller Is the code checking anywhere if the autolink plugin is enabled and running? Or does this line just fail if autolink is not running?

Reply:
Yup it just fails. No need to check as checking is probably just as expensive as just trying and failing.

Reply:
/giphy fire and forget :)

Reply:
I'm not sure if it's about cost. This code is only executed once on startup. 

My concerns are mostly that admins might be confused on the warning method.

Reply:
I agree with @hanzei. Maybe we can add something to the error that says "Make sure the autolink plugin is installed in order to auto-create links to your Jira issues."